src/Illuminate/Filesystem/FilesystemAdapter.php

```diff
@@ -544,6 +563,33 @@ public function getAwsTemporaryUrl($adapter, $path, $expiration, $options)
         )->getUri();
     }
 
+    /**
+     * Get a signed URL for Aws S3 via Cloudfront.
+     *
+     * @param  string $path
+     * @param  \DateTimeInterface $expiration
+     * @param  array $options
+     * @return string
+     */
+    public function getAwsCloudfrontSignedUrl($path, $expiration, $options)
+    {
+        $client = new CloudFrontClient([
+            'version' => 'latest',
+            'region' => env('AWS_DEFAULT_REGION'),
+            'debug' => env('APP_DEBUG'),
+        ]);
+
+        $resourceUrl = env('AWS_CLOUDFRONT_PROTOCOL').'://'.env('AWS_CLOUDFRONT_DOMAIN').$path;
```

Comment:
You can't use `env` outside of config files as it will return null for cached configuration.